### DISCUSSION TITLE
Directus is great, I have a few suggestions after using it for a few days

### DISCUSSION THREAD

**Discussion by zhuyifang** [directus/directus#6966] [Feature Requests]
1. If the data in webhooks can support simple script editing, it will be very useful
2. Global language packs support custom coverage
3. Built-in SMS or email verification code service
4. The field default value supports random strings and random numbers with a length that can be set
5. More detailed documents. For example, the default language settings in the current documents are not detailed enough. It took me a lot of time to know that it is enough to modify the language items in the user profile.
6. default layout can be designed for each model
7. Each module can set the upload path for the pictures used separately

Several problems have been found so far：
1. When creating a many-to-one or one-to-many relationship between the newly created model and the system users table, an error is always reported, but some SQL statements lack one execution, such as the creation of fields, which makes the program incomplete (the program and the data structure are inconsistent) , I hope this type of operation has a rollback function, that is, after an error, roll back the data structure。
2. Lack of sufficient context when participating in community translation services

**Comment by zhuyifang** [comment]
For the file upload part, I hope it can be more powerful.
1. The default upload path can be set separately for files of different types
2. The upload path can be set separately in different data models
3. When using single or multiple file upload components, you can set the allowed file types

**Comment by liuxiaodong008008** [comment]
I really want SMS code sign up and sign in feature

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
